mesonbuild/mtest.py

```diff
@@ -1976,6 +2000,9 @@ def get_tests(self, errorfile: T.Optional[T.IO] = None) -> T.List[TestSerialisat
         tests = [t for t in self.tests if self.test_suitable(t)]
         if self.options.args:
             tests = list(self.tests_from_args(tests))
+        if self.options.slice:
+            our_slice, nslices = self.options.slice
+            tests = tests[our_slice::nslices]
```

Comment:
We require slices in the format 1/10 through 10/10 to select each slice (`nslices >= subslice > 0`). Indexing into that naively will skip the first test and slice 10 will repeat all tests from slice 0 other than the first test.

Reply:
> Indexing into that naively will skip the first test and slice 10 will repeat all tests from slice 0 other than the first test.

Indeed, I also found that a couple minutes ago while writing tests for this change. Fixed now.